Add io.github.Cookiiieee.WSelector

Cookiiieee opened the pull request:
- Add Flatpak manifest
- Add appdata.xml with metadata
- Add README with build instructions
- Add application screenshot
- Initial submission for review

<!-- ⚠️⚠️  Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️  -->

<!-- 💡 Please go to the preview tab to view the markdown below 💡 -->

### Please confirm your submission meets all the criteria

<!-- 💡 Please replace each `[ ]` with `[X]` when the step is complete 💡 -->

<!-- 💡 Please tick and write 'N/A' with a reason if a checklist item below is not applicable 💡 -->

- [X] Please describe the application briefly. WSelector is a modern GTK4/libadwaita application that allows users to browse and set wallpapers from Wallhaven.cc. It features a clean, responsive interface with support for searching, filtering, and previewing wallpapers before setting them.
 -->
- [X] Please include a screenrecord (10-20s) showcasing the application. https://www.reddit.com/r/gnome/comments/1kxg3yd/wselector_v10_a_modern_gtk4_wallpaper_selector/
- [X] The Flatpak ID follows all the rules listed in the [Application ID requirements][appid].
- [X] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2] and I agree to them.
- [X] I am an author/developer/upstream contributor to the project.
      If not, I contacted upstream developers about this submission. **Link:**

<!-- 💡 Please mention below the GitHub usernames of any additional maintainers needed (if any) 💡 -->

<!-- ⚠️⚠️  Please DO NOT modify anything below this line ⚠️⚠️  -->

[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission

hfiguiere reviewed (changes requested):
Please re-read the submission documentation and address the layout of the files in the submission.

hfiguiere commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,68 @@
+        "--socket=fallback-x11",
+        "--filesystem=home",
+        "--filesystem=xdg-pictures:create",
+        "--socket=system-bus",
+        "--socket=session-bus"
```
these d-bus socket are not allowed. you need figure out which names you need.

Cookiiieee replied:
Yeah, Dbus doesn't work so this can be removed.

hfiguiere commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,68 @@
+            ]
+        },
+        {
+            "name": "app",
```
use an explicit name. (app isn't)

Cookiiieee replied:
Noted, will make the change accordingly.

Cookiiieee replied:
Fixed in recent update

hfiguiere commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,68 @@
+                "./install-icons.sh ${FLATPAK_DEST}",
+                "install -Dm644 wselector.desktop ${FLATPAK_DEST}/share/applications/io.github.Cookiiieee.WSelector.desktop",
+                "install -Dm644 io.github.Cookiiieee.WSelector.appdata.xml ${FLATPAK_DEST}/share/metainfo/io.github.Cookiiieee.WSelector.appdata.xml",
+                "sed -i 's/org\\.example\\.WSelector/io.github.Cookiiieee.WSelector/g' ${FLATPAK_DEST}/share/applications/io.github.Cookiiieee.WSelector.desktop",
```
NEVER use `sed` to edit desktop file. Why is this wrong the desktop file?

Cookiiieee replied:
For whatever reason I was having issues with the desktop file creation. Not if it's related to my desktop.ini or how flathub builds it... this was the only method I found that worked in my Linux environment. 

Noticed without this method, desktop file would get created but when clicking it nothing happens app doesn't load.

Open to suggestions!

hfiguiere commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,68 @@
+                "mkdir -p ${FLATPAK_DEST}/bin",
+                "mkdir -p ${FLATPAK_DEST}/lib/python3.12/site-packages/wselector",
+                "mkdir -p ${FLATPAK_DEST}/share/icons/hicolor/{16x16,32x32,48x48,64x64,96x96,128x128,192x192,256x256,512x512,scalable}/apps",
+                "mkdir -p ${FLATPAK_DEST}/share/applications",
+                "mkdir -p ${FLATPAK_DEST}/share/metainfo",
```
`install -D` create these for you
```suggestion
```

Cookiiieee replied:
Fixed in recent update

hfiguiere commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,68 @@
+        },
+        {
+            "name": "app",
+            "buildsystem": "simple",
```
this is very complex. you should have a build system to do that upstream.

Cookiiieee replied:
It doesn't require a meson build, works as it is now with simple.

hfiguiere commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,68 @@
+                    "path": "screenshots/screenshot.png"
+                },
+                {
+                    "type": "dir",
```
never use `dir` source. Also how does it find the actually source? This will not be run on your PC.

Cookiiieee replied:
Yeah, this is only used for the readme file, doesn't need to be included in actual build.

hfiguiere commented on io.github.Cookiiieee.WSelector.appdata.xml:
```diff
@@ -0,0 +1,53 @@
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://raw.githubusercontent.com/Cookiiieee/flathub/wselector-new-pr/screenshots/screenshot.png</image>
```
screenshot also should be upstream.

hfiguiere commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,65 @@
+            "sources": [
+                {
+                    "type": "file",
+                    "url": "https://raw.githubusercontent.com/Cookiiieee/WSelector/main/data/io.github.Cookiiieee.WSelector.appdata.xml",
```
never should source be pulled from the tip of a branch. Also a sha256 is required.

Cookiiieee replied:
Fixed in latest update.

hfiguiere commented:
> > The manifest MUST be in the top level of the repository.
> > Not sure which doc you followed but it is wrong.
> 
> Okay, I think I know where I went wrong, apologise for that! I've resubmitted the changes!

so which doc was that?

Cookiiieee commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,68 @@
+                    "https_proxy": "",
+                    "no_proxy": ""
+                },
+                "build-args": ["--share=network"]
```
Build requires network access to scrape wallpapers from wallhaven.cc website, would you have a recommended method I can use instead for "--share=network"?

bbhtt replied:
There's not method, not exceptions. Also please don't mark it as resolved when it is not.

Cookiiieee commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,68 @@
+        "--share=network",
+        "--socket=wayland",
+        "--socket=fallback-x11",
+        "--filesystem=home",
```
Home access stores the downloaded wallpapers, this is also the place where wallpapers will download to and can easily be managed by non-technical users.

Cookiiieee commented on io.github.Cookiiieee.WSelector.appdata.xml:
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>io.github.Cookiiieee.WSelector</id>
+  <metadata_license>GPL-3.0-or-later</metadata_license>
```
that I did not know about, I've updated the AppData file with the correct metadata to include CCO.

Cookiiieee commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,68 @@
+    "runtime-version": "48",
+    "sdk": "org.gnome.Sdk",
+    "command": "wselector",
+    "finish-args": [
```
Thank you for spotting that!

bbhtt replied:
i don't see it added

Cookiiieee commented:
Key Improvements:

- Moved appdata.xml to the main repository (now sourced from the release tarball)
- Removed all temporary scripts and simplified the build process
- Updated permissions to be more specific and secure
- Improved documentation and installation instructions
- Fixed all reported issues with desktop integration

The application has been thoroughly tested and is now ready for review further review. Thank you for your time and feedback!

hfiguiere reviewed (changes requested):
before requesting a review make sure everything is addressed.

bbhtt commented on io.github.Cookiiieee.WSelector.appdata.xml:
```diff
@@ -0,0 +1,53 @@
+    <category>GTK</category>
+    <category>Utility</category>
+  </categories>
+  <content_rating type="oars-1.1">
+    <content_attribute id="violence-cartoon">none</content_attribute>
+    <content_attribute id="violence-fantasy">none</content_attribute>
+    <content_attribute id="violence-realistic">none</content_attribute>
+    <content_attribute id="violence-bloodshed">none</content_attribute>
+    <content_attribute id="violence-sexual">none</content_attribute>
+    <content_attribute id="sex-nudity">none</content_attribute>
+    <content_attribute id="sex-themes">none</content_attribute>
+    <content_attribute id="language-profanity">none</content_attribute>
+    <content_attribute id="language-humor">none</content_attribute>
+    <content_attribute id="alcohol-drugs">none</content_attribute>
+    <content_attribute id="gambling">none</content_attribute>
+    <content_attribute id="discrimination">none</content_attribute>
+  </content_rating>
```
```suggestion
  <content_rating type="oars-1.1" />
```

bbhtt commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,73 @@
+        "--talk-name=org.freedesktop.portal.Desktop",
+        "--system-talk-name=org.freedesktop.Notifications",
+        "--talk-name=org.gnome.SessionManager",
+        "--talk-name=org.gnome.SettingsDaemon.MediaKeys",
```
```suggestion
```

this does not exist since gnome 42

bbhtt commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,73 @@
+        "--filesystem=xdg-data/wselector:create",
+        "--filesystem=xdg-pictures/wallpapers:ro",
+        "--talk-name=org.freedesktop.portal.Desktop",
+        "--system-talk-name=org.freedesktop.Notifications",
```
```suggestion
        "--talk-name=org.freedesktop.Notifications",
```

Cookiiieee replied:
this has been fixed to go through portal now

bbhtt commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,73 @@
+        "--filesystem=xdg-pictures/wallpapers:ro",
+        "--talk-name=org.freedesktop.portal.Desktop",
+        "--system-talk-name=org.freedesktop.Notifications",
+        "--talk-name=org.gnome.SessionManager",
```
what is this for?

bbhtt commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,73 @@
+        "--system-talk-name=org.freedesktop.Notifications",
+        "--talk-name=org.gnome.SessionManager",
+        "--talk-name=org.gnome.SettingsDaemon.MediaKeys",
+        "--talk-name=org.gtk.Actions",
```
what is this for?

Cookiiieee replied:
Updated to this now:

--share=network
Required for downloading wallpapers from Wallhaven.cc

--socket=wayland
Required for display on Wayland-based desktops

--share=ipc
Required for GTK4/libadwaita functionality

--socket=fallback-x11
For X11 compatibility on older systems

--socket=session-bus
Required for GTK file dialogs and desktop integration

--filesystem=xdg-pictures/WSelector:create
Required to save wallpapers to Pictures/WSelector

--talk-name=org.freedesktop.Notifications
For download/error notifications

--talk-name=org.freedesktop.portal.settings
For system theme detection (dark/light mode)

--talk-name=org.freedesktop.FileManager1
For "Show in folder" functionality

--talk-name=org.gtk.Actions
Required for GTK4 applications

bbhtt commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,73 @@
+            "buildsystem": "simple",
+            "build-commands": [
+                "mkdir -p ${FLATPAK_DEST}/bin",
+                "# Create necessary directories",
```
```suggestion
```

bbhtt commented on screenshots/screenshot.png:
this should not be here

Cookiiieee replied:
This has been removed now.

bbhtt commented on io.github.Cookiiieee.WSelector.appdata.xml:
not fixed

Cookiiieee commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,73 @@
+            "build-commands": [
+                "mkdir -p ${FLATPAK_DEST}/bin",
+                "# Create necessary directories",
+                "mkdir -p ${FLATPAK_DEST}/lib/python3.12/site-packages/wselector",
```
This has been updated.

hfiguiere commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,71 @@
+    "command": "wselector",
+    "build-options": {
+        "build-args": [
+            "--share=network"
```
how many times shall we tell that the build MUST be offline?

Cookiiieee replied:
Would it be okay to host the dependencies in github where users pull from that or directly from pythonhosted packages?

hfiguiere commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,71 @@
+            "name": "python-deps",
+            "buildsystem": "simple",
+            "build-commands": [
+                "python3 -m pip install --prefix=${FLATPAK_DEST} --no-binary :all: urllib3 charset-normalizer certifi idna requests beautifulsoup4 psutil"
```
where does it magically find the python packages? No online since the build MUST be offline. There is no network access.

Cookiiieee replied:
Working on that now, apologise for pushing the latest .json as I fixed a few things up from my original build.

hfiguiere commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,71 @@
+                "./install-icons.sh ${FLATPAK_DEST}",
+                "update-desktop-database ${FLATPAK_DEST}/share/applications",
+                "install -D /dev/null ${FLATPAK_DEST}/bin/wselector",
+                "echo '#!/bin/sh' > ${FLATPAK_DEST}/bin/wselector",
```
I said previously (https://github.com/flathub/flathub/pull/6558#discussion_r2113038114): NEVER create a script from build commands.

hfiguiere replied:
I though I was pretty clear about "a script" and not "this script", with echo.

![image](https://github.com/user-attachments/assets/3c82b81b-d488-443b-8155-d2e07bcb9f42)

There is a source of type "script".

Cookiiieee replied:
Oh don't you worry, next update coming soon! :D

hfiguiere commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,71 @@
+            ]
+        }
+    ],
+    "finish-args": [
```
`finish-args` and `cleanup` must be above `modules`.

hfiguiere commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,71 @@
+                "chmod +x install-icons.sh",
+                "./install-icons.sh ${FLATPAK_DEST}",
+                "update-desktop-database ${FLATPAK_DEST}/share/applications",
+                "install -D /dev/null ${FLATPAK_DEST}/bin/wselector",
```
```suggestion
```

Not what this is for.

Cookiiieee commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,73 @@
+        "--filesystem=xdg-config/wselector:create",
+        "--filesystem=xdg-data/wselector:create",
+        "--filesystem=xdg-pictures/wallpapers:ro",
+        "--talk-name=org.freedesktop.portal.Desktop",
```
as per my last comment, For system theme detection (dark/light mode)

Cookiiieee commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,73 @@
+        "--filesystem=xdg-pictures/WSelector:create",
+        "--filesystem=xdg-cache/wselector:create",
+        "--filesystem=xdg-config/wselector:create",
+        "--filesystem=xdg-data/wselector:create",
```
This has been fixed now

Cookiiieee commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,71 @@
+        "--socket=wayland",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=session-bus",
```
I thought you were referring to ' "--socket=system-bus",'  My app uses D-Bus to communicate with org.freedesktop.portal.Desktop for setting wallpapers via Flatpak. 

This is why --socket=session-bus was requested; essentially I want my app to work across multiple desktop environments. Limiting it to just gsettings will most likely cause limitations and falling back to xdg-desktop-portal without session bus access will break it for all non-Gnome desktops, which kind of defeats the purpose of having a wallpaper manager app released if it's limited to just gnome users.

hfiguiere replied:
It applied to both. I used the plural form. Not sure where it was unclear that you should determine which dbus names you need.

![image](https://github.com/user-attachments/assets/c3a10b7d-d789-4fe3-8974-ee71c6280dbf)

Cookiiieee replied:
finish-args-arbitrary-dbus-access
The [finish-args](https://docs.flatpak.org/en/latest/manifests.html#finishing) in the manifest has --socket=session-bus or --socket=system-bus.

This must not be used except for very specific cases. Please follow the [Flatpak permission guide](https://docs.flatpak.org/en/latest/sandbox-permissions.html#d-bus-access) for more information and find out the specific DBus names required using --log-session-bus.

this is a specific case, so I'm not sure why either is not allowed.

hfiguiere replied:
no. you still haven't explained why you need blanket session bus access.

" Required for GTK file dialogs and desktop integration" isn't a justification, because it doesn't need it.

Cookiiieee replied:
I will release a new manifest soon as I fix all the other issues mentioned in comments.

hfiguiere commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,71 @@
+                "install -Dm644 wselector.desktop ${FLATPAK_DEST}/share/applications/io.github.Cookiiieee.WSelector.desktop",
+                "install -Dm644 io.github.Cookiiieee.WSelector.metainfo.xml ${FLATPAK_DEST}/share/metainfo/io.github.Cookiiieee.WSelector.metainfo.xml",
+                "mkdir -p ${FLATPAK_DEST}/lib/python${PYVER}/site-packages",
+                "cp -r wselector ${FLATPAK_DEST}/lib/python${PYVER}/site-packages/",
```
If only python had facilities to install python modules...

hfiguiere commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,71 @@
+        "--talk-name=org.freedesktop.Notifications",
+        "--talk-name=org.freedesktop.portal.settings",
+        "--talk-name=org.freedesktop.FileManager1",
+        "--talk-name=org.gtk.Actions"
```
I'm not even sure what this is. There is no dbus with the name `org.gtk.Actions`

hfiguiere commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,71 @@
+        "*.a",
+        "*.pyc",
+        "*.pyo",
+        "__pycache__",
+        "**/__pycache__",
+        "**/*.pyc",
+        "**/*.pyo"
```
```suggestion
        "__pycache__"
```

Cookiiieee commented:
Hope the new changes meets requirements!

bbhtt commented on apps/io.github.Cookiiieee.WSelector/io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,204 @@
+        "/share/man",
+        "/include",
+        "/lib/*.la",
+        "/lib/*.a",
+        "/lib/python3.12/site-packages/*.pyc",
+        "/lib/python3.12/site-packages/*.pyo",
+        "/lib/python3.12/site-packages/__pycache__"
```
```suggestion
        "/lib/*.a"
```

Cookiiieee replied:
What's the problem here exactly?

bbhtt commented:
halfway through the review I realised you added a second manifest under `apps/` directory. maybe for once you can stop blindly following LLM's advice

the entire PR is a mess.

Cookiiieee commented on apps/io.github.Cookiiieee.WSelector/io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,204 @@
+        "--socket=fallback-x11",
+        "--share=network",
+        "--filesystem=xdg-pictures/WSelector:create",
+        "--talk-name=org.freedesktop.portal.Desktop",
+        "--talk-name=org.freedesktop.portal.Wallpaper",
```
Why is that the case?

Cookiiieee commented on apps/io.github.Cookiiieee.WSelector/io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,204 @@
+        "--filesystem=xdg-pictures/WSelector:create",
+        "--talk-name=org.freedesktop.portal.Desktop",
+        "--talk-name=org.freedesktop.portal.Wallpaper",
+        "--talk-name=org.kde.plasmashell"
```
Access to pictures/WSelector folder, also added functionality to apply wallpaper between kde. Unless im missing something it would be much appreciated if you can maybe not give me one line comments and actually let me know what the problem is exactly. Not everyone publishing an app has years of experience

Cookiiieee replied:
This is the only folder on the home directory that gets created and used to download and manage wallpapers

Cookiiieee commented:
Should I delete PR and start from scratch

Cookiiieee commented:
Okay, to summarise since I've changed a few things. 

1. I've deleted old branches and will only be using this branch from now on 'wselector-new-pr'. 

git branch -a
  master
  new-pr
* wselector-new-pr
  remotes/fork/HEAD -> fork/master
  remotes/fork/master
  remotes/fork/wselector-new-pr
  remotes/origin/HEAD -> origin/master
  remotes/origin/bbhtt/merge-fixes
  remotes/origin/master
  remotes/origin/new-pr
  
2. I've updated the "Cleanup" portion of the build. 
  
Kept: /share/man
Reason: Manual (man) pages.
Why: These are only needed for development or debugging, not for end-users.

Kept: /include
Reason: Header files (.h and similar) used for compiling C/C++ code.
Why: End-users don’t need these; they’re only for development.

Kept: /lib/*.la
Reason: Libtool archive files (.la).
Why: These are only needed for building from source, not for running the app.

Kept: /lib/*.a
Reason: Static library files (.a).
Why: Used for static linking during builds, not needed at runtime.

Kept: /lib/python3.12/site-packages/__pycache__
Reason: Python bytecode cache directories.
Why: These can be safely regenerated if needed, and removing them reduces bundle size.

3. Justification for my finish-args. 

## Filesystem: 
--filesystem=xdg-pictures/WSelector:create
Reason: This is needed for the wallpaper directory (~/Pictures/WSelector). Wallpapers are downloaded, stored and managed from this folder making it easily accessible from the home directory. No other folders are used to store wallpapers.

## Portals:
--talk-name=org.freedesktop.portal.Desktop
Reason: This is required to (view "wallpaper directory").
--talk-name=org.freedesktop.portal.Wallpaper 
Reason: Setting wallpaper.
--talk-name=org.kde.plasmashell
Reason: Needed for interacting with KDE Plasma's wallpaper system.

hfiguiere commented:
Why did you merge master into this?

![image](https://github.com/user-attachments/assets/df79deed-1d46-413b-b694-633079f52843)

Cookiiieee commented:
> Why did you merge master into this?
> 
> ![image](https://private-user-images.githubusercontent.com/114441/452658080-df79deed-1d46-413b-b694-633079f52843.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkzMzk3NTUsIm5iZiI6MTc0OTMzOTQ1NSwicGF0aCI6Ii8xMTQ0NDEvNDUyNjU4MDgwLWRmNzlkZWVkLTFkNDYtNDEzYi1iNjk0LTYzMzA3OWY1Mjg0My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYwN1QyMzM3MzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMmRjNDZkZDJlZGFkMzZkMGI1MjM4NWJhYmQ4YTNjMTBlODQ5OGNlNGE2NDQ5NWY2NWIyOTBkYjMyMjc1ZDU0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.b5fOt9K5HaPD3dEj0Iqy_8nSSazshrkj7DIgVEADcQQ)

This is now fixed.

Cookiiieee commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,202 @@
+        ]
+    }
+    ],
+    "cleanup": [
```
This has now been done

Cookiiieee commented:
Let us know if there are any other changes required

bbhtt commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,201 @@
+        "--socket=fallback-x11",
+        "--share=network",
+        "--filesystem=xdg-pictures/WSelector:create",
+        "--talk-name=org.freedesktop.portal.Wallpaper",
```
```suggestion
```

Again you don't use the portal by adding permissions.

bbhtt commented on io.github.Cookiiieee.WSelector.json:
```diff
@@ -0,0 +1,201 @@
+        "/share/man",
+        "/include",
+        "/lib/*.la",
+        "/lib/*.a",
+        "/lib/python3.12/site-packages/__pycache__"
```
```suggestion
        "/lib/*.a"
```

Cookiiieee commented:
Checked build failure, discovered my flatpak didn't have the install desktop file arg in the build command. Updating manifest and tested working from my main repo.

Cookiiieee commented:
Run docker run --rm \
+ flatpak-builder-lint --exceptions repo repo
Traceback (most recent call last):
  File "/app/bin/flatpak-builder-lint", line 10, in <module>
    sys.exit(main())
             ^^^^^^
  File "/app/lib/python3.12/site-packages/flatpak_builder_lint/cli.py", line 230, in main
    if results := run_checks(
                  ^^^^^^^^^^^
  File "/app/lib/python3.12/site-packages/flatpak_builder_lint/cli.py", line 90, in run_checks
    check_method(check_method_arg)
  File "/app/lib/python3.12/site-packages/flatpak_builder_lint/checks/catalogue.py", line [20](https://github.com/flathub-infra/vorarbeiter/actions/runs/15575897153/job/43860677132#step:14:21)4, in check_repo
    self._validate(tmpdir, appid, ref_type)
  File "/app/lib/python3.12/site-packages/flatpak_builder_lint/checks/catalogue.py", line 132, in _validate
    if not re.match(rf"^{appid}([-.].*)?[.]desktop$", launchable[0]):
                                                      ~~~~~~~~~~^^^
IndexError: list index out of range
error: Recipe `validate-build` failed with exit code 1
Error: Process completed with exit code 1.

**Note** 
Need to look into this issue as to why it's failing during build validation.

Cookiiieee commented:
Checked appdata and discovered launchable is missing. Added "<launchable type="desktop-id">io.github.Cookiiieee.WSelector.desktop</launchable>" in my repo, I believe this should fix this launchable[0] error.

Cookiiieee commented:
bot, build

Cookiiieee commented:
re-attempting build with new build SHA256

Cookiiieee commented:
Latest test build log: 

+ flatpak-builder-lint --exceptions repo repo
{
    "errors": [
        "non-svg-icon-in-scalable-folder",
        "appstream-failed-validation"
    ],
    "warnings": [
        "appstream-screenshot-missing-caption"
    ],
    "appstream": [
        "E: io.github.Cookiiieee.WSelector:8: developer-id-missing"
    ],
    "info": [
        "appstream-failed-validation: Metainfo file /work/tmp/tmp54b3s977/metainfo/io.github.Cookiiieee.WSelector.metainfo.xml has failed validation. Please see the errors in appstream block",
        "non-svg-icon-in-scalable-folder: ['/work/tmp/tmpk9r53med/icons/hicolor/scalable/apps/io.github.Cookiiieee.WSelector.png']",
        "appstream-screenshot-missing-caption: One or more screenshots are missing captions in the Metainfo file"
    ],
    "message": "Please consult the documentation at https://docs.flathub.org/docs/for-app-authors/linter"
}
error: Recipe `validate-build` failed with exit code 1

Will investigate futher

bbhtt commented:
/merge head=4f3e278ec8dea64f30504a19802adba0660b1608
